java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -113,39 +114,56 @@ private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final
-      .filter(learnedConstraint -> learnedConstraint.getSv().equals(symbolicValue))
+      .filter(lc -> lc.getSv().equals(symbolicValue))
       .map(ExplodedGraph.Node.LearnedConstraint::getConstraint)
-      .peek(constraint -> flowForConstraint(currentNode, parent, constraint, selectedYield));
+      .peek(lc -> learnedConstraintFlow(lc, currentNode, parent).forEach(flow::add));
   }
 
-  @CheckForNull
-  private static MethodYield getMethodYieldFromParent(ExplodedGraph.Node currentNode, ExplodedGraph.Node parent) {
-    Tree syntaxTree = parent.programPoint.syntaxTree();
-    if (syntaxTree != null && syntaxTree.is(Tree.Kind.METHOD_INVOCATION)) {
-      return currentNode.selectedMethodYield(parent);
+  private Stream<JavaFileScannerContext.Location> learnedConstraintFlow(@Nullable Constraint learnedConstraint, ExplodedGraph.Node currentNode, ExplodedGraph.Node parent) {
+    if (learnedConstraint == null || !addToFlow.test(learnedConstraint)) {
+      return Stream.empty();
     }
-    return null;
-  }
-
-  private void flowForConstraint(ExplodedGraph.Node currentNode, ExplodedGraph.Node parent, @Nullable Constraint constraint, @Nullable MethodYield selectedMethodYield) {
-    if (constraint != null && addToFlow.test(constraint)) {
-      flow.add(location(parent, learnedConstraintMessage(constraint, currentNode, parent)));
-      if (selectedMethodYield != null) {
-        flow.addAll(flowFromMethodInvocation(selectedMethodYield, parent));
-      }
+    Tree nodeTree = parent.programPoint.syntaxTree();
+    if (nodeTree.is(Tree.Kind.METHOD_INVOCATION)) {
+      return methodInvocationFlow(learnedConstraint, currentNode, parent);
     }
+    if (nodeTree.is(Tree.Kind.NEW_CLASS)) {
+      return Stream.of(location(parent, String.format("Constructor implies '%s'.", learnedConstraint.valueAsString())));
+    }
+    String name = SyntaxTreeNameFinder.getName(nodeTree);
+    String message = name == null ? learnedConstraint.valueAsString() : String.format("Implies '%s' is %s.", name, learnedConstraint.valueAsString());
+    return Stream.of(location(parent, message));
   }
 
-  private List<JavaFileScannerContext.Location> flowFromMethodInvocation(MethodYield usedMethodYield, ExplodedGraph.Node parentNode) {
-    int argumentIndex = correspondingArgumentIndex(symbolicValue, parentNode);
-    return usedMethodYield.flow(argumentIndex);
+  private Stream<JavaFileScannerContext.Location> methodInvocationFlow(Constraint learnedConstraint, ExplodedGraph.Node currentNode, ExplodedGraph.Node parent) {
+    MethodInvocationTree mit = (MethodInvocationTree) parent.programPoint.syntaxTree();
+    MethodYield selectedMethodYield = currentNode.selectedMethodYield(parent);
+    if (selectedMethodYield == null) {
+      return Stream.of(location(parent, String.format("'%s()' invoked.", mit.symbol().name())));
+    }
+    Stream.Builder<JavaFileScannerContext.Location> flow = Stream.builder();
+    if (selectedMethodYield.isExceptional()) {
```

Comment:
This is necessary for `UnclosedResourcesCheck` to function properly, because rule raises the issue on place where SV has opened status. The rule will replace the default message with it's own version ("resource is opened") , however in cases when the rule is not raising the issue the message was "...".  I replaced it with more explanatory text.